lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1303,8 +1285,10 @@ def _initialize_security_index(self, admin_secrets: Dict[str, any]) -> None:
     def _get_nodes(self, use_localhost: bool) -> List[Node]:
         """Fetch the list of nodes of the cluster, depending on the requester."""
         # This means it's the first unit on the cluster.
-        if self.unit.is_leader() and not self.peers_data.get(
-            Scope.APP, "security_index_initialised", False
+        if (
+            self.unit.is_leader()
+            and "data" in self.opensearch_peer_cm.deployment_desc().config.roles
```

Comment:
should the first unit of the main orchestrator (when dedicated `cluster_manager`) also receive `[]` from this method?

Reply:
I think that would result in a break of the workflow. Currently we try to get the online nodes, if not possible, we defer (see [here](https://github.com/canonical/opensearch-operator/blob/main/lib/charms/opensearch/v0/opensearch_base_charm.py#L940-L951), logic was not changed with this PR).

If we change that and return an empty list, we would instead raise a `OpenSearchNotFullyReadyError` and not defer / come back. Then the cluster-manager would never fully start up, and the later-starting data node could not join the cluster after initializing the security index.